datafusion/physical-expr-common/src/physical_expr.rs

```
@@ -144,6 +145,26 @@ pub trait PhysicalExpr: Send + Sync + Display + Debug + DynEq + DynHash {
         Ok(Some(vec![]))
     }
 
+    /// Finds the final statistic of the expression by combining the input statistics
+    /// in a post-order bottom-up manner (post-order DFS in statistics graph).
+    /// 
+    /// The part of [`StatisticsV2`] framework, work in progress.
+    fn evaluate_statistics(&self, _stats: &[&StatisticsV2]) -> Result<StatisticsV2> {
+        not_impl_err!("Not implemented for {self}")
+    }
+
+    /// Updates children statistic, having a known parent statistic for this expression.
+    /// This is used to propagate constraints down through an expression tree.
+    /// 
+    /// The part of [`StatisticsV2`] framework, work in progress.
+    fn propagate_statistics(
+        &self,
+        _parent_stat: &StatisticsV2,
+        _children_stat: &[&StatisticsV2],
+    ) -> Result<Option<Vec<StatisticsV2>>> {
+        Ok(Some(vec![]))
```

I know this is the logic that used in propagate_constraints --returning an empty vector instead of cloning `children` if none of the children intervals change as a result of propagation-- should we leave this behavior for simplicity and expressiveness? I am also thinking that changing the propagate_constraints in this manner. I think this hack is bringing more debug or code complexity cost than its non-clone efficiency

It's not a big deal but I'd like to discuss it.

---

> returning an empty vector instead of cloning children if none of the children intervals change as a result of propagation

The original option with empty vec is safer from OOP point of view. The complete solution should be implemented like you've mentioned, but **all** expressions should be revisited.